numpy/core/tests/test_umath.py

```diff
@@ -692,6 +692,25 @@ def assert_complex_equal(x, y):
                 assert_complex_equal(np.power(zero, -p), cnan)
             assert_complex_equal(np.power(zero, -1+0.2j), cnan)
 
+    # Testing 0^{Non-zero} issue 18378
+    def test_zero_power_nonzero(self):
+        zero = np.array([0j])
+        cnan = np.array([complex(np.nan,np.nan)])
+
+        def assert_complex_equal(x, y):
+            assert_array_equal(x.real, y.real)
+            assert_array_equal(x.imag, y.imag)
+
+        with np.errstate(invalid="ignore"):
+            #real part and imaginary part both greater than zero
+            assert_complex_equal(np.power(zero,1+4j),zero)
+            #real part greater than zero imag part less than zero
+            assert_complex_equal(np.power(zero,2-3j),zero)
+            #real part less than zero imag part greater than zero
+            assert_complex_equal(np.power(zero,-1+1j),cnan)
+            #real part less than zero imag part less than zero
+            assert_complex_equal(np.power(zero,-2-3j),cnan)
```

Comment:
Can you split these out and use `pytest.warns()`? That way it is explicitly about which case gives warnings and which does not. (and the test will notify it if it does not work properly due to a change or weird platform)